sentry_sdk/scope.py

```diff
@@ -1250,3 +1421,55 @@ def __repr__(self):
             hex(id(self)),
             self._name,
         )
+
+
+@contextmanager
+def new_scope():
+    # type: () -> Generator[Scope, None, None]
+    """
+    Context manager that forks the current scope and runs the wrapped code in it.
+
+    .. versionadded:: X.X.X
+    """
+    current_scope = Scope.get_current_scope()
+    forked_scope = current_scope.fork()
+    token = _current_scope.set(forked_scope)
+
+    try:
+        yield forked_scope
+
+    finally:
+        # restore original scope
+        _current_scope.reset(token)
+
+
+@contextmanager
+def isolated_scope():
```

Comment:
~~If I read the RFC right this should be copy-on-write~~ Nevermind, I guess that's coming in the next PR. :)

Reply:
yes, copy on write stuff is in second PR.